File: fsharp/raindrops/Raindrops.fs
module Raindrops

let convert n = 
    let numberToText (d, t) =
        if n % d = 0
        then Some t
        else None

    let textsOrInputNumber = function
        | [] -> n.ToString()
        | xs -> String.concat "" xs

    [(3, "Pling"); (5, "Plang"); (7, "Plong")]
    |> List.map numberToText
    |> List.choose id
    |> textsOrInputNumber

File: csharp/raindrops/Raindrops.cs
using System.Text;

public static class Raindrops
{
    public static string Convert(int number)
    {
        var sb = new StringBuilder();
        if (number % 3 == 0) sb.Append("Pling");
        if (number % 5 == 0) sb.Append("Plang");
        if (number % 7 == 0) sb.Append("Plong");
        if (sb.Length < 1) sb.Append(number.ToString());
        return sb.ToString();
    }
}